models/vrp.rb

```diff
@@ -268,19 +269,37 @@ def self.convert_position_relations(hash)
       relations_to_remove.reverse_each{ |index| hash[:relations].delete_at(index) }
     end
 
+    def self.deduce_first_solution_strategy(hash)
+      preprocessing = hash[:configuration] && hash[:configuration][:preprocessing]
+      return unless preprocessing
+
+      # To make sure that internal vrp conforms with the Grape vrp.
+      preprocessing[:first_solution_strategy] ||= []
+      preprocessing[:first_solution_strategy] = [preprocessing[:first_solution_strategy]].flatten
+    end
+
+    def self.deduce_minimum_duration(hash)
+      resolution = hash[:configuration] && hash[:configuration][:resolution]
+      return unless resolution && resolution[:initial_time_out]
+
+      log 'initial_time_out and minimum_duration parameters are mutually_exclusive', level: :warn if resolution[:minimum_duration]
```

Comment:
can this happen out of tests?

Reply:
I guess not :thinking: 

I thought the best place for this warning (which is mainly for development) is here since we do the replacement here and once the `initial_time_out` is completely deprecated, this function will be removed and the `log` will be deleted automatically with it.

Reply:
You are right, since this is only temporary I guess we can keep it like that.
We could eventually move it to _create_ function in _test_helper.rb_. Would that work ?

Reply:
we need to keep the `initial_time_out` -> `minimum_duration` replacement here as long as `initial_time_out` stays on the api so it is better to keep the log together I think.